src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -310,11 +310,15 @@ namespace smt::noodler {
         std::vector<std::pair<size_t, Predicate>> regs;
         this->formula.get_side_regulars(regs);
         std::deque<std::pair<size_t, Predicate>> worklist(regs.begin(), regs.end());
+        // keeps the ids of removed equations
+        std::set<size_t> removed;
 
         while(!worklist.empty()) {
             std::pair<size_t, Predicate> pr = worklist.front();
             worklist.pop_front();
 
+            if (removed.contains(pr.first)) continue; // if the equation was already removed, we do not remove it again
```

Comment:
How this can happen? There are two equal equations?

Reply:
For equation `x=y`, both sides "become regular" by removing other equations containing `x` and `y`.

Reply:
It happened because there was some equation `u = v` and the function detecting regular subparts realised that both `u` and `v` are regular, so it added two equations `regular1 = u`, `regular2 = v` and the original equation transformed into `regular1 = regular2`.